packages/cli/src/cli/server/action.server.ts

```diff
@@ -22,37 +28,46 @@ const BaseProvider: ConfigProvider = {
 
 const DefaultPort = 5000;
 
-export class BasemapsServerCommand extends BaseCommandLine {
-  config = this.defineStringParameter({
-    argumentName: 'CONFIG',
-    parameterLongName: '--config',
-    description: 'Configuration source to use',
-  });
-  ignoreConfig = this.defineFlagParameter({
-    parameterLongName: '--no-config',
-    description: 'Assume no config and just load tiffs from the configuration path',
-  });
-  bundle = this.defineStringParameter({
-    argumentName: 'BUNDLE',
-    parameterLongName: '--bundle',
-    description: 'Compile the configuration into a bundle and output path',
-  });
-  assets = this.defineStringParameter({
-    argumentName: 'ASSETS',
-    parameterLongName: '--assets',
-    description: 'Where the assets (sprites, fonts) are located',
-  });
-  port = this.defineIntegerParameter({
-    argumentName: 'PORT',
-    parameterLongName: '--port',
-    description: 'port to use',
-    defaultValue: DefaultPort,
-  });
-
-  constructor() {
+export class CommandBasemapsServer extends CommandLineAction {
+  config: CommandLineStringParameter;
+  ignoreConfig: CommandLineFlagParameter;
+  bundle: CommandLineStringParameter;
+  assets: CommandLineStringParameter;
+  port: CommandLineIntegerParameter;
+
+  protected onDefineParameters(): void {
+    this.config = this.defineStringParameter({
```

Comment:
I think this is much easier not to put this here, and just to use the the previious `config = this.defineStringParam`

as there is no need to double define the types.

Reply:
Previously we are extend `BaseCommandLine` a instead of `CommandLineAction`, as we need combine these actions together I have to convert it to `CommandLineAction`, and onDefineParameters() must be defined in this class.